src/modal/alert-modal.interface.ts

```diff
@@ -46,8 +46,9 @@ export interface AlertModalData {
 	buttons?: Array<ModalButton>;
 	/**
 	 * Callback for non-specific close events. `return false;` to prevent the modal from closing
+	 * Property trigger has a value of `overlay` or `close`
 	 */
-	close?: Function;
+	close?: (trigger) => boolean;
```

Comment:
We'll want to add the possible types here

```suggestion
	close?: (trigger: "overlay" | "close") => boolean;
```